### ISSUE TITLE
Can't change Bevel if mosaic is selected

### ISSUE THREAD

**Issue by Eustache84** [azagaya/laigter#34]
Hi there ! 

Your soft is perfect for what I need ( I can generate normal map from a very large source, like 15000x10000px ) 
But I have an issue : 

When I check Mosaic for tileable patern, i can't change anymore bevel features ( height, distance .. ) it's not blank but nothing happen when I change these settings, 
When I try with the native pic, it's ok, but with mine ( even if I just create a square with paint ) I can't... 

Here is my setup : 
win7
Ryzen 1700x
40go DDR4
660gtx 
Laigter 1.8.1
..

Thanks :+1:

**Comment by azagaya** [comment]
Hi! thanks for the report.
Are you talking about the checkbox by the neighbours button? If so, the behaviour you are describing is in fact the expected behaviour. That checkbox is used in textures that should be able to be "tiled". When the checkbox is checked, Laigter will "extend" the texture in all directions, and use the resulting image to calculate the normals. This is useful for textures like this one:
 
![imagen](https://user-images.githubusercontent.com/46932830/74271859-5fb65880-4cec-11ea-8d64-648efc94b322.png)

If the checkbox is unchecked, the normal map generated is this:

![imagen](https://user-images.githubusercontent.com/46932830/74271967-8ffdf700-4cec-11ea-92d2-5d09e50ec723.png)

But with the checkbox enabled, is this:

![imagen](https://user-images.githubusercontent.com/46932830/74272018-a906a800-4cec-11ea-98c0-c0f7e295c0ea.png)

So, when making tiles of the texture ingame, first option would look like this:

![tiletest-p](https://user-images.githubusercontent.com/46932830/74272100-cdfb1b00-4cec-11ea-8c48-b6c9a803ec42.png)

And the tiles wont be seamless... with the checkbox checked you get this result instead:
![tiletest-p2](https://user-images.githubusercontent.com/46932830/74272174-ebc88000-4cec-11ea-884a-09bd73f7ac83.png)

Wich correctly generates seamless tiles. You can also use the "neighbours" to select how the texture should be extended when using the mosaic checkbox, for using for example with tilemaps.

In summary, if your texture is completely filled (i.e no fully transparent zones) to the edges, that checbox will prevent bevel functions, as it will assume that is a tile and so it doesn't have edges. The bevel functions only works for edges of a texture (i mean, where texture is fully transparent).

I dont know if i made myself clear, please tell me if i wasn't, and tell me what are you trying to achieve. You can join us on discord for a more dynamic help also.

**Comment by Eustache84** [comment]
Wow ! Thanks for the quick answer !
My english is not so good, IDK if I've understand all what you said but here is my case. 

My picture is an tilable texture of a stone wall so it can be repeated in each side without seem. 
There is no blank or transparent part ( just stone and black part like this : https://thumbs.dreamstime.com/b/fond-de-mur-en-pierre-19561367.jpg ) 

I think this is why it doesn't work, 
I'm looking to spread the bevel of each stones to make it less rough but without changing the height. 
So it's probably not an issue, just misunderstanding ! 

Thanks again 
( i've no microphone so i can't discord with you ;) )

**Comment by azagaya** [comment]
Hi!

> ( i've no microphone so i can't discord with you ;) )

We can chat on discord also.. i never use spoken channels. But cahtting lets you a more dynamic response!

> I'm looking to spread the bevel of each stones to make it less rough but without changing the height.

Oh, so i think what you are talking about is something that is usually caled "preserve large details", or "large scale". I have not implemented such a feature yet, because it was never requested unitl now. There is another tool called [NormalMapGenerator](https://github.com/Theverat/NormalmapGenerator) that has it implemented, although it doesnt have the exact same purpose than laigter. If you try it and find thats what you want, **i may be able to implement it in laigter**.

Another thing you can try using more smooth on the enhance controls, but this wall also smooth the details, leaving you something like this:
![walltest](https://user-images.githubusercontent.com/46932830/74330887-47d4e800-4d71-11ea-8baa-871dcc7d01c5.png)

**Comment by azagaya** [comment]
There is also another thing you could try, but it involves a bit more of editting work. You could make a copy of your image, open it in your edditting soft (gimp, photoshop, etc), and erase all the sites inbetween the rocks. Save it as another file and load it as a heightmap (rightclick in the texture in the texture list, and click load heightmap). This way you could use the bevel controls. I tried with this eddited image ask heightmap:
![wallmask](https://user-images.githubusercontent.com/46932830/74332547-bd8e8300-4d74-11ea-8813-443a40ef1770.png)
And got this result:
![ezgif com-video-to-gif(2)](https://user-images.githubusercontent.com/46932830/74332584-d303ad00-4d74-11ea-8665-d2abd0cfbc46.gif)

**Comment by azagaya** [comment]
@Eustache84 were my comments useful? May i close this?

**Comment by azagaya** [comment]
I'm closing this because apparently no more help is needed. I can reopen it if needed.
